File: JanperCab/3-Application/Dtos/Process/ProcessDto.cs
using _1_Domain.Enum;
using _3_Application.Dtos.OnHoldComponent;
using System;
using System.Collections.Generic;

namespace _3_Application.Dtos.Process
{
    public abstract class ProcessDto
    {
        public int Id { get; set; }

        public ProcessTypeEnum ProcessType { get; set; }

        public int EnquiryId { get; set; }

        public DateTime? StartTime { get; set; }

        public DateTime? EndTime { get; set; }

        public bool IsCurrent { get; set; }


        public ICollection<OnHoldComponentDto> OnHoldComponents { get; set; }
    }
}

File: JanperCab/5-JanperCab/ClientApp/src/app/_models/DuraformProcess/ProcessDto.ts
import { OnHoldComponentDto } from './../on-hold-detail/OnHoldComponentDto';
import { Type } from 'class-transformer';

export abstract class ProcessDto {
  $type: string;
  id: number;
  enquiryId: number;
  startTime: Date;
  endTime: Date;
  isCurrent: boolean;

  @Type(() => OnHoldComponentDto)
  onHoldComponents: OnHoldComponentDto[];
}
